src/blas/KokkosBlas_unmqr.hpp

```diff
@@ -0,0 +1,257 @@
+  }
+  if (!(valid_trans)) {
+    std::ostringstream os;
+    os << "KokkosBlas::unmqr: trans[0] = '" << trans[0] << "'. "
+       << "Valid values include 'T' or 't' (Transpose), 'N' or 'n' (No "
+          "transpose).";
+    Kokkos::Impl::throw_runtime_exception(os.str());
+  }
+
+  int64_t A0   = A.extent(0);  // M if 'L', N if 'R'
+  int64_t A1   = A.extent(1);  // > k
+  int64_t C0   = C.extent(0);  // M
+  int64_t C1   = C.extent(1);  // N
+  int64_t tau0 = tau.extent(0);
+
+  // Check validity of Tau
+  if (tau0 < k) {
+    std::ostringstream os;
+    os << "KokkosBlas::unmqr: Dimensions of tau and k do not match (require "
+          "len(tau) >=k ): "
+       << "k: " << k << "Tau: " << tau0;
+    Kokkos::Impl::throw_runtime_exception(os.str());
+  }
+
+  // Check validity of k
+  if ((side[0] == 'L') || (side[0] == 'l')) {
+    if ((k > C0) || (k < 0)) {
+      std::ostringstream os;
+      os << "KokkosBlas::unmqr: Number of reflectors k must not exceed M. "
+         << "M: " << C0 << " "
+         << "k: " << k;
+      Kokkos::Impl::throw_runtime_exception(os.str());
+    }
+    if ((A0 != C0)) {
+      std::ostringstream os;
+      os << "KokkosBlas::unmqr: A must be of size M x k: "
+         << "A: " << A0 << " x " << A1 << " "
+         << "M: " << C0;
+      Kokkos::Impl::throw_runtime_exception(os.str());
+    }
+  } else {
+    if ((k > C1) || (k < 0)) {
+      std::ostringstream os;
+      os << "KokkosBlas::unmqr: Number of reflectors k must not exceed N. "
+         << "N: " << C1 << " "
+         << "k: " << k;
+      Kokkos::Impl::throw_runtime_exception(os.str());
+    }
+    if ((A0 != C1)) {
+      std::ostringstream os;
+      os << "KokkosBlas::unmqr: A must be of size N x k: "
+         << "A: " << A0 << " x " << A1 << " "
+         << "N: " << C1;
+      Kokkos::Impl::throw_runtime_exception(os.str());
+    }
+  }
+#endif  // KOKKOSKERNELS_DEBUG_LEVEL > 0
+
+  // return if degenerate matrix provided
+  if ((A.extent(0) == 0) || (A.extent(1) == 0)) return;
+  if ((C.extent(0) == 0) || (C.extent(1) == 0)) return;
+  if ((k == 0)) return;
+
+  // standardize particular View specializations
+  typedef Kokkos::View<
+      typename AViewType::const_value_type**, typename AViewType::array_layout,
+      typename AViewType::device_type, Kokkos::MemoryTraits<Kokkos::Unmanaged> >
+      AVT;
+
+  typedef Kokkos::View<typename TauViewType::const_value_type*,
+                       typename TauViewType::array_layout,
+                       typename TauViewType::device_type,
+                       Kokkos::MemoryTraits<Kokkos::Unmanaged> >
+      TVT;
+
+  typedef Kokkos::View<typename CViewType::non_const_value_type**,
+                       typename CViewType::array_layout,
+                       typename CViewType::device_type,
+                       Kokkos::MemoryTraits<Kokkos::Unmanaged> >
+      CVT;
+
+  typedef Kokkos::View<typename WViewType::non_const_value_type*,
```

Comment:
All these views are unmanaged. Why?

Reply:
I was following the style given by other functions in the BLAS wrappers: See https://github.com/kokkos/kokkos-kernels/blob/master/src/blas/KokkosBlas3_gemm.hpp#L232.  Honestly, I'm not sure why they would need to be. There shouldn't be any worry of them going out of scope.